budvm/src/lib.rs

```diff
@@ -2186,6 +2214,8 @@ where
             })?,
         };
 
+        self.stack.pop_n(to_pop);
```

Comment:
This function can exit on errors early, do we need to clean up the stack in these cases?